pkg/assembler/backends/neo4j/pkg.go

```diff
@@ -246,3 +253,210 @@ func (e *nameToVersion) PropertyNames() []string {
+			}
+			if pkgSpec.Name != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "name", "name", "$pkgName")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "name", "name", "$pkgName")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgName"] = pkgSpec.Name
+			}
+			if pkgSpec.Version != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "version", "version", "$pkgVerion")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "version", "version", "$pkgVerion")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgVerion"] = pkgSpec.Version
+			}
+
+			if pkgSpec.Subpath != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "version", "subpath", "$pkgSubpath")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "version", "subpath", "$pkgSubpath")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["pkgSubpath"] = pkgSpec.Subpath
+			}
+
+			if pkgSpec.MatchOnlyEmptyQualifiers != nil && !*pkgSpec.MatchOnlyEmptyQualifiers {
+
+				if len(pkgSpec.Qualifiers) > 0 {
+
+					for _, qualifier := range pkgSpec.Qualifiers {
+						// neo4j does not accept "." in its properties. If the qualifier contains a "." that must
+						// be replaced by an "-"
+						qualifierKey := strings.ReplaceAll(qualifier.Key, ".", "_")
+						if firstMatch {
+							err := matchWhere(&sb, "version", qualifierKey, "$"+qualifierKey)
+							if err != nil {
+								return nil, fmt.Errorf("string builder failed with err: %w", err)
+							}
+							firstMatch = false
+						} else {
+							err := matchAnd(&sb, "version", qualifierKey, "$"+qualifierKey)
+							if err != nil {
+								return nil, fmt.Errorf("string builder failed with err: %w", err)
+							}
+						}
+						queryValues[qualifierKey] = qualifier.Value
+					}
+				}
+			}
+
+			sb.WriteString(" RETURN type.type, namespace.namespace, name.name, version")
+
+			result, err := tx.Run(sb.String(), queryValues)
+			if err != nil {
+				return nil, err
+			}
+
+			pkgTypes := map[string]map[string]map[string][]*model.PackageVersion{}
+			pkgNamespaces := map[string]map[string][]*model.PackageVersion{}
+			pkgNames := map[string][]*model.PackageVersion{}
+			for result.Next() {
+
+				versionNode := result.Record().Values[3].(dbtype.Node)
+
+				pkgQualifiers := []*model.PackageQualifier{}
+				if pkgSpec.MatchOnlyEmptyQualifiers != nil && !*pkgSpec.MatchOnlyEmptyQualifiers {
+					if len(pkgSpec.Qualifiers) > 0 {
```

Comment:
I think either way these 2 paths are equivalent, perhaps we can just keep the else block?